src/util/map.js

```diff
@@ -146,3 +146,9 @@ export function merge<K, V>(
   }
   return result;
 }
+
+export function pushValue<K, V>(map: Map<K, V[]>, key: K, value: V): void {
```

Comment:
This definitely needs docs.

What does it do?

Does it guarantee that the identity of an existing array value is
preserved?